src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -300,6 +300,14 @@ def _build_os_profile():
         if secrets:
             os_profile['secrets'] = secrets
 
+        if enable_auto_update is not None and custom_image_os_type.lower() == 'windows':
```

Comment:
How about combine these 2 `custom_image_os_type.lower() == 'windows'` ?

Reply:
I group code by parameter. It is not a serious problem.